src/reanimated2/js-reanimated/JSReanimated.ts

```diff
@@ -78,6 +83,12 @@ export default class JSReanimated {
     _iosReferenceFrame: number,
     eventHandler: ShareableRef<(data: Value3D | ValueRotation) => void>
   ): number {
+    if (!isWindowAvailable()) {
+      // the window object is unavailable when building the server portion of a site that uses SSG
+      // this check is here to ensure that the server build won't fail
+      return -1;
+    }
+
```

Comment:
Should be added to `initalizeSensor` as well.

Reply:
I see two ways, add a check to `initializeSensor` and `detectPlatform` or mark them as `private`. I'm more convinced to mar them as a private. I'll create another PR with proper encapsulation for the whole file.